modules/helm/Makefile

```diff
@@ -25,9 +25,15 @@ helm/toolbox/upsert:
 
 ## Delete all releases in a `NAMEPSACE` as well as the namespace
 helm/delete/namespace:
-	@helm list --namespace $(NAMESPACE) --short | xargs -r helm delete --purge
+	$(call assert-set,NAMESPACE)
+	@$(HELM) list --namespace $(NAMESPACE) --short | xargs -r $(HELM) delete --purge
 	@kubectl delete namespace $(NAMESPACE) --ignore-not-found
 
+## Delete `NAMEPSACE` if there are no releases in it
+helm/delete/namespace/empty:
+	$(call assert-set,NAMESPACE)
+	@[ "$(shell $(HELM) list --namespace $(NAMESPACE) --short --all)" ] || kubectl delete namespace $(NAMESPACE) --ignore-not-found
```

Comment:
I think you should be explicit here. Use `-n` for "non-empty" string.

Reply:
Don't use `$(shell ...)` because it will *always* evaluate, not just when the target is called. Remember, a `Makefile` is a template that is preprocessed just like a helmfile.

Reply:
I do not know how to do that without `shell`